standards/groundhog_standard.md

```diff
@@ -0,0 +1,235 @@
+```SOLIDITY
+
+/** @dev returns the value of the subscription
+  * @param bytes _subscriptionHash is the identifier of the customer's subscription with its relevant details.
+  * @return data is the data that has is required as apart of the txn for the requested subscription
+*/
+function getSubscriptionData(bytes _subscriptionHash) view public returns (bytes data);
+```
+
+###### getSubscriptionInterval
+```SOLIDITY
+
+/** @dev returns the interval of the subscription 
+  * @param bytes _subscriptionHash is the identifier of the customer's subscription with its relevant details.
+  * @return interval is the interval for re-billing of the requested subscription
+*/
+function getSubscriptionInterval(bytes _subscriptionHash) view public returns (uint interval);
+```
+
+###### getSubscriptionExternalId
+```SOLIDITY
+
+/** @dev returns the value of the subscription
+  * @param bytes _subscriptionHash is the identifier of the customer's subscription with its relevant details.
+  * @return externalId is a hex encoded external identifier for the customers subscription
+*/
+function getSubscriptionExternalId(bytes _subscriptionHash) view public returns (uint externalId);
+
+```
+
+###### getSubscriptionDestination
+```SOLIDITY
+
+/** @dev returns the value of the subscription
+  * @param bytes _subscriptionHash is the identifier of the customer's subscription with its relevant details.
+  * @return destination is the value that has been subscribed for the requested subscription
+*/
+function getSubscriptionDestination(bytes _subscriptionHash) view public returns (address destination);
+
+```
+
+###### getSubscriptionRecipient
+```SOLIDITY
+
+/** @dev returns the value of the subscription
+  * @param bytes _subscriptionHash is the identifier of the customer's subscription with its relevant details.
+  * @return recipient is the address that the subscription pays out to.
+*/
+function getSubscriptionRecipient(bytes _subscriptionHash) view public returns (address recipient);
+```
+
+###### getSubscriptionExpire
+```SOLIDITY
+
+/** @dev returns the value of the subscription
+  * @param bytes _subscriptionHash is the identifier of the customer's subscription with its relevant details.
+  * @return expires is the timestamp that this subscription can no longer be re-billed
+*/
+function getSubscriptionExpire(bytes _subscriptionHash) view public returns (uint expires);
+
+```
+
+###### getSubscriptionStatus
+```SOLIDITY
+
+/** @dev returns the value of the subscription
+  * @param bytes _subscriptionHash is the identifier of the customer's subscription with its relevant details.
+  * @return status is the enumerated status of the current subscription, 0 expired, 1 active, 2 paused, 3 cancelled
+*/
+function getSubscriptionStatus(bytes _subscriptionHash) view public returns (uint status);
+
+```
+#### Public Functions
+
+
+###### createSubscription
+```SOLIDITY
+
+/** @dev handles the incoming data to define subscription rules
+  * @param address _destination
```

Comment:
i think that `_data`, and `_meta` needs more specification to say what it is.  what should be stored here?

for `expires` and `_interval`, are these unix timestamps and ints that represent seconds, respectively?